core/build.gradle

```diff
@@ -142,7 +142,7 @@ task integrationTestCosmos(type: Test) {
         systemProperties(System.getProperties())
     }
     maxParallelForks = 3
-    jvmArgs '-XX:MaxDirectMemorySize=2g'
+    jvmArgs '-XX:MaxDirectMemorySize=2g', '-Xmx4g'
```

Comment:
OOM error occurred in the Cosmos Integration test after this change. So increased the heap size.